components/pages/game/components/Game.tsx

```diff
@@ -1,27 +1,109 @@
-// import React, { useEffect, useState } from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 
-import ImageType from 'domain/Image';
+import { RESULT } from 'config/routes';
+import DefaultTemplate from 'components/shared/templates/DefaultTemplate';
+import TimerBar from 'components/shared/atoms/TimerBar';
+import useEndGame from 'lib/apollo/hooks/actions/endGame';
+import useSendAnswerAndGetNextQuestion from 'lib/apollo/hooks/actions/sendAnswerAndGetNextQuestion';
+import { useRouter } from 'next/router';
+import GameStep from './GameStep';
+import HeaderChildren from './HeaderChildren';
+import { TitleDescription } from './styled';
 
-// import GameStep from './GameStep';
-
-interface StartGameQuestion {
-  answerOptions: string;
+interface Question {
+  answerOptions: string[];
   avatarUrl: string;
 }
 
-const Game = ({ images, initialQuestion }: { images: ImageType[]; initialQuestion: StartGameQuestion }) => {
-  // const [step, setStep] = useState(0);
-  // const [currentQuestion, setCurrentQuestion] = useState(initialQuestion);
-  // const [answer, setAnswer] = useState('');
-  // useEffect(() => {
-  //   {mutate, mutationState} = useAddAnswerGetQuestion();
-  //   await mutate(answer)
-  // }, [answer]);
-  // const handleAddAnswer = () => {
-  //   setAnswer(answer)
-  // };
-  // return currentQuestion ? <GameStep question={currentQuestion} addAnswer={handleAddAnswer} /> : <>Нет вопросов</>;
-  return null;
+const FULL_BAR = 100;
+const FULL_TIME = 45;
+
+function useTimer(limit: number, cb = console.log): number {
+  const [time, setTime] = useState(limit);
+  const onceCallRef = useRef(false);
+  if (time <= 0 && onceCallRef.current === false) {
+    cb();
+    onceCallRef.current = true;
+  }
+  useEffect(() => {
+    function tick() {
+      setTime((prevTime) => (prevTime === 0 ? 0 : prevTime - 1));
+    }
+    const id = setInterval(tick, 1000);
+    return () => clearInterval(id);
+  }, [time]);
+  return time;
+}
+
+const Game = ({ initialQuestion, gameId }: { initialQuestion: Question; gameId: string }) => {
+  const [currentQuestion, setCurrentQuestion] = useState<Question>(initialQuestion);
+  const [currentAnswer, setCurrentAnswer] = useState<string | undefined>(undefined);
+
+  const [sendAnswerAndGetNextQuestion, { data, loading, error }] = useSendAnswerAndGetNextQuestion();
+  const [endGame] = useEndGame();
+
+  const router = useRouter();
+
+  const isCurrentAnswerCorrect =
```

Comment:
predicat can't be null

Reply:
it is not predicate. It's three different mutually exclusive values